server/archive.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const lsArchive = require('ls-archive');
+
+function isArchive(path) {
+    return lsArchive.isPathSupported(path);
+}
+
+function listArchive(path) {
+    return new Promise((resolve, error) => {
+        try {
+            lsArchive.list(path, (err, entries) => {
+                if (err) {
```

Comment:
I think there is no need in `try-catch` and `Promise` constructor. Let's use [util.promisify](https://nodejs.org/dist/latest-v15.x/docs/api/util.html#util_util_promisify_original) instead:

```js
const {promisify} = require('util');
const list = promisify(lsArhive.list);

module.exports.listArchive = (path) => {
    const [error, result] = await tryToCatch(list, path);
};
```

Let's use [try-to-catch](https://github.com/coderaiser/try-to-catch) here.